### PR TITLE
Use submatches to replace strings with specific prefixes

### PR DESCRIPTION
**Pull request by kzys**
This change is for handling hcsshim that has RuntimeNs and
KernelTime_100Ns. The former has to be RuntimeNS, but the latter should
be intact.

Signed-off-by: Kazuyoshi Kato <katokazu@amazon.com>

### CODE REVIEW

File: cmd/go-fix-acronym/rewrite.go [hunk 1]
```diff
@@ -37,10 +37,34 @@ func rewriteNode(pattern *regexp.Regexp, node ast.Node) {
...
+			matches := pattern.FindAllStringSubmatchIndex(ident.Name, -1)
+			if matches == nil {
+				return true
+			}
+
+			var match []int
+			for _, match = range matches {
+				// No submatches.
+				if len(match) == 2 {
+					result += ident.Name[0:match[0]]
```
Comment (dmcgowan) [inline cmd/go-fix-acronym/rewrite.go:52]: This doesn't behave right when there are multiple matches. I had to try for myself

```.diff
diff --git a/cmd/go-fix-acronym/rewrite_test.go b/cmd/go-fix-acronym/rewrite_test.go
index 7bc65ab..2966b1f 100644
--- a/cmd/go-fix-acronym/rewrite_test.go
+++ b/cmd/go-fix-acronym/rewrite_test.go
@@ -59,6 +59,12 @@ func TestRewriteSimple(t *testing.T) {
                        expected: "//hello\npackage main\n\nfunc GetCPU() {}",
                        c:        config{acronyms: []string{"Cpu"}},
                },
+               {
+                       name:     "simple",
+                       input:    "//hello\npackage main\nfunc GetCpuFromCpuList(){}",
+                       expected: "//hello\npackage main\n\nfunc GetCPUFromCPUList() {}",
+                       c:        config{acronyms: []string{"Cpu"}},
+               },
                {
                        name: "group",
                        input: `package main
```

```
$ go test ./cmd/go-fix-acronym
--- FAIL: TestRewriteSimple (0.00s)
    --- FAIL: TestRewriteSimple/simple#01 (0.00s)
        rewrite_test.go:45: expected "//hello\npackage main\n\nfunc GetCPUFromCPUList() {}\n", but got "//hello\npackage main\n\nfunc GetCPUGetCpuFromCPUList() {}\n"
FAIL
FAIL	github.com/containerd/protobuild/cmd/go-fix-acronym	0.004s
FAIL
```

Comment (kzys) [reply cmd/go-fix-acronym/rewrite.go]: Thanks. It should be fixed in the last revision.

File: cmd/go-fix-acronym/rewrite.go [hunk 2]
```diff
@@ -37,10 +37,31 @@ func rewriteNode(pattern *regexp.Regexp, node ast.Node) {
...
+				}
+
+				for i := 0; i < len(match); i += 2 {
+					result += ident.Name[begin:match[i]]
+					result += strings.ToUpper(ident.Name[match[i]:match[i+1]])
+					begin = match[i+1]
+				}
+			}
+			// Copy the rest.
+			result += ident.Name[match[1]:len(ident.Name)]
```
Comment (dmcgowan) [inline cmd/go-fix-acronym/rewrite.go:63]: Should this just use `begin`? The `1` index would be confusing if for some reason there were multiple submatches

Comment (kzys) [reply cmd/go-fix-acronym/rewrite.go:63]: Sounds good. Will do.

**Review by dmcgowan — APPROVED** [review]

**Pull request merged**